example/lib/presentation/state/custom_clusterization_map_strategy.dart

```diff
@@ -0,0 +1,214 @@
+          icon: PlacemarkIcon.single(
+            PlacemarkIconStyle(
+              image: BitmapDescriptor.fromAssetImage(Assets.cluster),
+            ),
+          ),
+          consumeTapEvents: true,
+          text: PlacemarkText(
+            text: cluster.pointsSize.toString(),
+            style: const PlacemarkTextStyle(
+              color: Colors.black,
+              size: 10,
+            ),
+          ),
+          opacity: 1,
+          mapId: MapObjectId(cluster.clusterId!.toString()),
+        );
+      }
+
+      final id = int.parse(cluster.markerId!);
+
+      return PlacemarkMapObject(
+        point: cluster.location,
+        icon: PlacemarkIcon.single(
+          PlacemarkIconStyle(
+            image: BitmapDescriptor.fromAssetImage(
+              _selectedIds.contains(id) ? Assets.routeStart : Assets.routeEnd,
+            ),
+          ),
+        ),
+        consumeTapEvents: true,
+        opacity: 1,
+        mapId: MapObjectId(cluster.markerId!),
+        onTap: (_, __) {
+          if (_selectedIds.contains(id)) {
+            _selectedIds.remove(id);
+          } else {
+            _selectedIds.add(id);
+          }
+
+          notifyListeners();
+        },
+      );
+    }).toSet();
+  }
+
+  @override
+  String get title => "Custom clusterization";
+
+  late final List<ClusterableWrapper> _clusterables;
+
+  late final Fluster<ClusterableWrapper> _fluster;
+
+  CustomClusterizationMapStrategyDelegate() {
+    /// Generate clusterables objects,
+    /// which are used for Fluster tp
+    /// generate clusters.
+    _clusterables = _generateClusterables();
+
+    /// Create Fluster instance.
+    ///
+    /// Fluster is a helper class to generate
+    /// clusters for the map.
+    ///
+    /// It uses a quadtree data structure to
+    /// efficiently cluster large amounts of
+    /// points.
+    _fluster = Fluster<ClusterableWrapper>(
+      minZoom: 0,
+      maxZoom: 20,
+      radius: 150,
+      extent: 2048,
+      nodeSize: 64,
+      points: _clusterables,
+      // ignore: avoid_types_on_closure_parameters
+      createCluster: (BaseCluster? cluster, double? longitude, double? latitude) {
+        return ClusterableWrapper(
+          latitude: latitude,
+          longitude: longitude,
+          isCluster: true,
+          clusterId: cluster?.id,
+          pointsSize: cluster?.pointsSize,
+          childMarkerId: cluster?.childMarkerId,
+        );
+      },
+    );
+  }
+
+  List<ClusterableWrapper> _generateClusterables() {
+    final clusterables = <ClusterableWrapper>[];
+
+    for (int i = 0; i < 100; i++) {
+      clusterables.add(
+        ClusterableWrapper(
+          latitude: Constants.defaultLocation.latitude + (0.1 * (i - 50)),
+          longitude: Constants.defaultLocation.longitude + (0.1 * (i - 50)),
+          isCluster: false,
+          clusterId: 0,
+          pointsSize: 0,
+          markerId: '$i',
+        ),
+      );
+    }
+    return clusterables;
```

Comment:
minor - you can use `List.generate` here

Reply:
done